lib/portlayer/exec/exec.go

```diff
@@ -151,6 +154,22 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 			log.Errorf("Error encountered during container cache sync during init process: %s", err)
 			return
 		}
+
+		if Config.UseVMGroup {
+			vmGroupChan := make(chan chan error, batchingLimit)
+			Config.addToVMGroup = func() error {
+				errChan := make(chan error)
+				vmGroupChan <- errChan
+				err := <-errChan
```

Comment:
technically this should be a `select` with a case for `op->Done()` so that cancellation works on this side as well. The `op` would need to be passed into `addToVMGroup` as an argument.